### PR TITLE
MNT update version references post v0.7.0 release

### PR DESCRIPTION
**Pull request by romanlutz**
This is the usual update for all mentions of the current version according to the release guide.
Additionally, this changes which versions are shown on the website. Instead of using just wildcards we will only publish docs for latest patch version of every minor version (excluding 0.6.0 and 0.6.1 since 0.6.2 exists, for example). This is basically implementing the approach proposed in https://github.com/fairlearn/fairlearn/discussions/659#discussioncomment-766177 since it seemed to resonate with people (or at least @hildeweerts and @riedgar-ms). 

@fairlearn/fairlearn-maintainers

Signed-off-by: Roman Lutz <rolutz@microsoft.com>

### CODE REVIEW

**Review by riedgar-ms — APPROVED** [review]

File: README.md
Comment (adrinjalali) [inline README.md:23]: here we could instead refer users to the `pypi.org` page which would show them the latest release by default, or our website which directs users to the latest release. And not change it here with every release.

File: docs/static_landing_page/index.html
Comment (adrinjalali) [inline docs/static_landing_page/index.html:30]: would it be possible to make this a variable and change it once in this file?

**Review by adrinjalali — COMMENTED** [review]
thanks @romanlutz

File: docs/static_landing_page/index.html
Comment (romanlutz) [reply docs/static_landing_page/index.html]: Yes! Done!

**Comment by romanlutz** [comment]
@adrinjalali can you take another look? This should be fine now and v0.7.0 is out so we can merge this if it's acceptable.

File: README.md
Comment (adrinjalali) [inline README.md:1]: does this have to be a single line?

File: docs/conf.py
Comment (adrinjalali) [inline docs/conf.py:97]: could we change this to a regex accepting any versions?

**Review by adrinjalali — COMMENTED** [review]
nitpicks, otherwise LGTM, thanks @romanlutz

File: README.md
Comment (romanlutz) [reply README.md:1]: I believe so. Otherwise they're on several lines. One day we'll be as fancy as scikit-learn and convert it into a README.rst where we have more flexibility 🙂

Comment (adrinjalali) [reply README.md:1]: I see, ok.

File: docs/conf.py
Comment (romanlutz) [reply docs/conf.py:97]: Yes, but I consciously don't want to accept all versions, only the highest patch version of each minor (see PR description). Maybe that's possible to do with a regex (?)
However, I felt like it's easier/safer/more intuitive to write a little bit of code to generate this regex (added in last couple commits). Wdyt?

File: README.md
Comment (romanlutz) [reply README.md:1]: Thanks for opening #884, I'll wait for that to be merged and make the changes in the new README.rst 🙂

Comment (romanlutz) [reply README.md:1]: This is now much easier to read because we migrated to README.rst, so I'll resolve this comment.

File: docs/conf.py
Comment (romanlutz) [reply docs/conf.py]: The PR includes the extra code now.

File: docs/conf.py
Comment (adrinjalali) [inline docs/conf.py:126]: you probably want to use `packaging.version.parse` and friends here (https://stackoverflow.com/a/11887885/2536294)

File: README.rst
Comment (MiroDudik) [inline README.rst:30]: How standard / accepted is `badge.fury.io`? I'd be more comfortable linking to https://pypi.org/project/fairlearn/

File: README.rst
Comment (MiroDudik) [inline README.rst:166]: Again, i'm not sure why not link to https://pypi.org/project/fairlearn/

**Review by MiroDudik — APPROVED** [review]
I just have two small quibbles with links, but I'm okay if this gets in... e.g., if there is some functionality that would be lost by providing pypi.org link, then let's keep this as is.

File: README.rst
Comment (romanlutz) [reply README.rst:30]: Somehow I wasn't able to get a pypi link pointing at the latest version, but yours does! Not sure how I didn't find that one 🙂 I'm replacing it in the next commit.

File: README.rst
Comment (romanlutz) [reply README.rst:166]: Same as above, replacing this shortly 🙂 Thanks!

File: docs/conf.py
Comment (romanlutz) [reply docs/conf.py:126]: Switching to `packaging` in the next commit! Thanks for the suggestion.

File: docs/static_landing_page/js/landing_page.js
Comment (kurianbenoy) [inline docs/static_landing_page/js/landing_page.js:176]: No newline at EOF

**Comment by romanlutz** [comment]
Fixed everything now and the versions are all showing up as expected:
image

@adrinjalali or @hildeweerts can you merge this? That's assuming the gates pass, of course.

**Comment by hildeweerts** [comment]
@romanlutz it seems that 'main' has disappeared since I've merged this - that 's not what was expected right?
image

**Comment by romanlutz** [comment]
> @romanlutz it seems that 'main' has disappeared since I've merged this - that 's not what was expected right?
> 
> image
> 
> 

Not at all expected! Thanks for merging and I'll look into it right away.

**Pull request merged**
